engine.js

```diff
@@ -193,6 +197,133 @@ var accumulateResult = function(ifResult, thenResult) {
         DEBUG = level;
     };
 
+    HMDAEngine.getDebug = function() {
+        return DEBUG;
+    };
+
+    /*
+     * -----------------------------------------------------
+     * Local DB
+     * -----------------------------------------------------
+     */
+
+    HMDAEngine.setUseLocalDB = function(bool) {
+        _USE_LOCAL_DB = bool;
+        if (bool) {
+            return resetDB();
+        } else {
+            return destroyDB();
+        }
+    };
+
+    HMDAEngine.shouldUseLocalDB = function() {
+        return _USE_LOCAL_DB;
+    };
+
+    HMDAEngine.loadCensusData = function(year) {
+        var apiCalls = [
+            this.apiGET('localdb/census/msaCodes'),
+            this.apiGET('localdb/census/stateCounty'),
+            this.apiGET('localdb/census/stateCountyMSA'),
+            this.apiGET('localdb/census/stateCountyTract'),
+            this.apiGET('localdb/census/stateCountyTractMSA')
+        ];
+        return Promise.map(apiCalls, function(body) {
+            return loadDB(resultFromResponse(body));
+        });
+    };
+
+    var resetDB = function() {
+        return destroyDB()
+        .then(function() {
+            _LOCAL_DB = levelup('hmda');
+            return _LOCAL_DB;
+        });
+    };
+
+    var destroyDB = function() {
+        var deferred = Promise.defer();
+
+        var realDestroy = function() {
+            if (typeof levelup.destroy === 'function') {
```

Comment:
`levelup` proxies it when the backing storage exposes it, like in the case of using `leveldown` in node. Hence the checking if it's a function..